student-store-ui/src/components/App/App.jsx

```diff
@@ -1,20 +1,132 @@
 import * as React from "react"
+// import {BrowserRouter} from 'react-router-dom';
 import Navbar from "../Navbar/Navbar"
 import Sidebar from "../Sidebar/Sidebar"
 import Home from "../Home/Home"
+import ProductDetail from "../ProductDetail/ProductDetail";
+import NotFound from "../NotFound/NotFound"
+import { BrowserRouter, Routes, Route, Link } from 'react-router-dom';
 import "./App.css"
 
+import { useState } from "react";
+import { useEffect } from "react";
+import axios from "axios";
 export default function App() {
+  const [products, setProductStat] = useState([]);
+  const [isFetching, setFetchingStat] = useState(false);
+  const [isOpen, setOpenStat] = useState(false);
+  const [shoppingCart, setShoppingCartStat] = useState([]);
+  const [checkoutForm, setCheckoutForm] = useState({ "name": "", "email": "" });
+  const [checkoutFormStat, setCheckoutFormStat] = useState(null);
+  const [receipt, setReceipt] = useState(null);
+  const [error, setErrorStat] = useState("");
+  const URL = "https://codepath-store-api.herokuapp.com/store";
+  useEffect(async () => {
+    const { data } = await axios(URL)
+      .catch(function (error) {
+        setLoading(true)
+      })
+    setProductStat(data.products)
+  }, [])
+
+  function handleOnToggle() {
+    setOpenStat(!isOpen);
+  };
+  function handleAddItemsToCart(productId) {
+    // TODO: It should add that product to the shoppingCart if
+    // it doesn't exist, and set its quantity to 1.
+    //  It should add the price of the product to the total
+    // price of the shoppingCart.
+    let temp = [...shoppingCart];
+    let inCart = false;
+
+    temp = temp.map((item) => {
+      if (item.itemId === productId) {
+        inCart = true;
+        return { ...item, quantity: item.quantity + 1 };
+      }
+      else {
+        return item;
+      }
+
+    }
+    );
+    if (!inCart) {
+      temp = [...temp, { itemId: productId, quantity: 1 }];
+    }
+    setShoppingCartStat(temp)
+
+  };
```

Comment:
to make the logic surrounding the shopping cart easier to access -> can instead store it as an object

{
   productID: { 
      quantity: x,
   }
}

then whenever need to iterate though for say rendering use Object.keys, Object.values, etc